tests/unit/test_remote_write.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2020 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import unittest
+from unittest.mock import patch
+
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
+from charms.prometheus_k8s.v0.prometheus_remote_write import (
+    PrometheusRemoteWriteConsumer,
+)
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+from charm import INGRESS_MULTIPLE_UNITS_STATUS_MESSAGE, Prometheus, PrometheusCharm
+
+RELATION_NAME = "receive-remote-write"
+RELATION_INTERFACE = "prometheus_remote_write"
+METADATA = f"""
+name: consumer-tester
+requires:
+  {RELATION_NAME}:
+    interface: {RELATION_INTERFACE}
+"""
+
+FILES = {}
+
+ALERT_RULES = {
+    "groups": [
+        {
+            "name": "None_f2c1b2a6-e006-11eb-ba80-0242ac130004_consumer-tester_alerts",
+            "rules": [
+                {
+                    "alert": "CPUOverUse",
+                    "expr": 'process_cpu_seconds_total{juju_model="None",'
+                    'juju_model_uuid="f2c1b2a6-e006-11eb-ba80-0242ac130004",'
+                    'juju_application="consumer-tester"} > 0.12',
+                    "for": "0m",
+                    "labels": {
+                        "severity": "Low",
+                        "juju_model": "None",
+                        "juju_model_uuid": "f2c1b2a6-e006-11eb-ba80-0242ac130004",
+                        "juju_application": "consumer-tester",
+                    },
+                    "annotations": {
+                        "summary": "Instance {{ $labels.instance }} CPU over use",
+                        "description": "{{ $labels.instance }} of job "
+                        "{{ $labels.job }} has used too much CPU.",
+                    },
+                },
+                {
+                    "alert": "PrometheusTargetMissing",
+                    "expr": 'up{juju_model="None",'
+                    'juju_model_uuid="f2c1b2a6-e006-11eb-ba80-0242ac130004",'
+                    'juju_application="consumer-tester"} == 0',
+                    "for": "0m",
+                    "labels": {
+                        "severity": "critical",
+                        "juju_model": "None",
+                        "juju_model_uuid": "f2c1b2a6-e006-11eb-ba80-0242ac130004",
+                        "juju_application": "consumer-tester",
+                    },
+                    "annotations": {
+                        "summary": "Prometheus target missing (instance {{ $labels.instance }})",
+                        "description": "A Prometheus target has disappeared."
+                        "An exporter might be crashed.\n"
+                        "VALUE = {{ $value }}\n  LABELS = {{ $labels }}",
+                    },
+                },
+            ],
+        }
+    ]
+}
+
+
+def fake_push(self, path, content, **kwargs):
```

Comment:
no longer needed

Reply:
Still needed from my testing. If you insist it's not, feel free to show me in a separate PR and we'll remove it.